### DISCUSSION TITLE
Stellarium Desktop DB and App Server

### DISCUSSION THREAD

**Discussion by IAASorg** [Stellarium/stellarium#3524] [Q&A]
Looking for advice for Stellarium...

I have several computers around my observatory that run Stellarium and it's a pain to keep all of them updated with the most recent DB/App versions.

Is there a setting I missed to allow the software to pull the DB and/or App from a central onsite server? I don't want to have to have each machine pinging the Internet each time it opens (running Stellarium web) or updates (running Stellarium desktop) due to external bandwidth limitations. If I could  run a local network server and use Stellarium pointing to that server rather than a locally installed copy on each machine that would be the ideal solution.

I also don't want to use an app server (like Citrix) as there are local (in observatory) devices that need to plug into a local machine to run the scope, camera, dome, etc.

I work with some local highschool students who use these observatories as well and ultimately I'd like to hand them the "keys" and let them take first level responsibility for the "simple" stuff like keeping the machines up to date.

**Comment by gzotti** [comment]
Stellarium (desktop) does not call home by itself to check for program updates. I also don't want that. There is no "in-program update" button.  It checks for updates in satellite or meteor shower data by the respective plugins and then fetches a few kB of data. You can set sources for the satellite elements (TLEs) to your central server, and make sure these TLEs are kept updated by other means (cronjob...) yourself. 

We cannot comment on what Stellarium Web does. 

You can direct Stellarium to use a User Data Directory different from default. See User Guide, 6.2.2. You could use the same directory for all instances of Stellarium to keep them in-sync. It is advisable to run Stellarium with dedicated config files (esp. when those computers are used for different topics) and log files for each running instance, though. See chapter 6 how to achieve this. You can also install DSS or even HiPS locally to save bandwidth. See 10.4.1. However, when updating solar system data or instrument data (Oculars plugin, TelescopeControl plugin), changes may be overwritten by whoever edited data last.

**Comment by xalioth** [comment]
Hi, stellarium web is a website, so it's not really possible to avoid a network connection. We tried hard to optimize caching of the files, but for some data like the satellites or for search queries the client needs to make an http connection.
